packages/vite/src/node/utils.ts

```diff
@@ -1192,6 +1192,45 @@ export const isNonDriveRelativeAbsolutePath = (p: string): boolean => {
   return windowsDrivePathPrefixRE.test(p)
 }
 
+/**
+ * Determine if a file is being requested with the correct case, to ensure
+ * consistent behaviour between dev and prod and across operating systems.
+ */
+export function shouldServe(url: string, assetsDir: string): boolean {
+  // viteTestUrl is set to something like http://localhost:4173/ and then many tests make calls
+  // like `await page.goto(viteTestUrl + '/example')` giving us URLs beginning with a double slash
+  const pathname = decodeURIComponent(
+    new URL(url.startsWith('//') ? url.substring(1) : url, 'http://example.com')
+      .pathname
+  )
```

Comment:
Looks like sirv is able to handle `//` because it uses `@polka/url`. We could probably use that too as it caches the result to `req`, but also something we can improve later.